### PR TITLE
feat(coinbase): add pagination in fetchLedger

### PR DESCRIPTION
**Pull request by sc0Vu**
fix: ccxt/ccxt#21756

Test:
```BASH
# should fetch all history
$  n coinbase fetchLedger USDT 1673986028000 100 '{"paginate": true,"account_id":"0e2b7cd9-ce67-53f3-92f2-d61c9410a9f6"}'
$  n coinbase fetchLedger USDT 1673986028000 1 '{"paginate": true,"account_id":"0e2b7cd9-ce67-53f3-92f2-d61c9410a9f6"}'

# should fetch single history
$  n coinbase fetchLedger USDT 1673986028000 1 '{"account_id":"0e2b7cd9-ce67-53f3-92f2-d61c9410a9f6"}'
```

### CODE REVIEW

File: ts/src/coinbase.ts [hunk 1]
```diff
@@ -1897,9 +1897,15 @@ export default class coinbase extends Exchange {
...
          * @param {int} [limit] max number of ledger entrys to return, default is undefined
          * @param {object} [params] extra parameters specific to the exchange API endpoint
+         * @param {boolean} [params.paginate] default false, when true will automatically paginate by calling this endpoint multiple times. See in the docs all the [availble parameters](https://github.com/ccxt/ccxt/wiki/Manual#pagination-params)
          * @returns {object} a [ledger structure]{@link https://docs.ccxt.com/#/?id=ledger-structure}
          */
         await this.loadMarkets ();
+        let paginate = false;
+        [ paginate, params ] = this.handleOptionAndParams (params, 'fetchLedger', 'paginate');
+        if (paginate) {
+            return await this.fetchPaginatedCallCursor ('fetchLedger', undefined, undefined, undefined, params, 'next_starting_after', 'starting_after', undefined, 100);
```
Comment (carlosmiei) [inline ts/src/coinbase.ts:1907]: I think we should add ` code, since, limit` here

**Comment by jtashiro** [comment]
I needed to modify coinbase.py fetch_accounts_v3() method to change the limit from 100 to 250 to get this to work properly.  When the fetch_ledger() method is called with only a currency code (not an accountId) the reverse lookup find_accounts() only has 100 elements in the list and what I'm seeing a full list of 157 items ...

### snippet of my python script
```
  params = {}
  limit = None
  ledger = exchange.fetch_ledger(code=code, since=since, limit=limit, params=params)
  print(f"# of ledger entries {len(ledger)}")
```

### example output with default of '100' limit in fetch_accounts_v3()
```
$ get-ledger.py --exchange coinbase --code "BTC" --since-date 2022-03-01 
CCXT Version: 4.2.76
ERROR: coinbase prepareAccountRequestWithCurrencyCode() could not find account id for BTC
```

### After changing lines 488, 490  to '250' this works ... 
```
        if paginate:
            return self.fetch_paginated_call_cursor('fetchAccounts', None, None, None, params, 'cursor', 'cursor', None, 250)
        request = {
            'limit': 250,
        }
```
### This is the working example output
```
$ get-ledger.py --exchange coinbase --code "BTC" --since-date 2022-03-01 
CCXT Version: 4.2.76
# of ledger entries 16

```

**Comment by jtashiro** [comment]
Bug this feature - if response is empty will fail in with index error

need to add

        if length == 0:
           return ledger

**Comment by jtashiro** [comment]
when I remove the length == 0 test, here is the stacktrace

```
  File "/Users/jtashiro/dev/exchange-trading/./get-ledger.py", line 197, in <module>
    ledger = exchange.fetch_ledger(code=code, since=since, limit=limit, params=params)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jtashiro/anaconda3/lib/python3.11/site-packages/ccxt/coinbase.py", line 1827, in fetch_ledger
    return self.fetch_paginated_call_cursor('fetchLedger', code, since, limit, params, 'next_starting_after', 'starting_after', None, 100)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jtashiro/anaconda3/lib/python3.11/site-packages/ccxt/base/exchange.py", line 5213, in fetch_paginated_call_cursor
    raise e
  File "/Users/jtashiro/anaconda3/lib/python3.11/site-packages/ccxt/base/exchange.py", line 5193, in fetch_paginated_call_cursor
    response = getattr(self, method)(symbol, since, maxEntriesPerRequest, params)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jtashiro/anaconda3/lib/python3.11/site-packages/ccxt/coinbase.py", line 1839, in fetch_ledger
    last = self.safe_dict(ledger, lastIndex)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jtashiro/anaconda3/lib/python3.11/site-packages/ccxt/base/exchange.py", line 1816, in safe_dict
    return self.safe_dict_n(dictionary, [key], defaultValue)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jtashiro/anaconda3/lib/python3.11/site-packages/ccxt/base/exchange.py", line 1803, in safe_dict_n
    value = self.safe_value_n(dictionaryOrList, keys, defaultValue)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jtashiro/anaconda3/lib/python3.11/site-packages/ccxt/base/exchange.py", line 884, in safe_value_n
    value = Exchange.get_object_value_from_key_list(dictionary, key_list)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jtashiro/anaconda3/lib/python3.11/site-packages/ccxt/base/exchange.py", line 894, in get_object_value_from_key_list
    if (key < len(dictionary_or_list)) and (dictionary_or_list[key] is not None) and (dictionary_or_list[key] != ''):
                                            ~~~~~~~~~~~~~~~~~~^^^^^
IndexError: list index out of range
```

**Comment by carlosmiei** [comment]
@jtashiro Thanks, will add your suggested fix

**Pull request merged**